src/auth/negotiate/kerberos/negotiate_kerberos_auth.cc

```diff
@@ -640,6 +642,18 @@ main(int argc, char *const argv[])
                 xfree(spnegoToken);
             }
             xfree(token);
+            xfree(rcache_type);
+            xfree(rcache_type_env);
```

Comment:
AFAICT, freeing putenv()-borrowed variables is unsafe because (compliant) putenv() implementations use the supplied memory "as is" instead of copying it. The safe options are:

1. Leak the value. Do not free it at all. Not recommended, of course.
1. Unset the environment variable before freeing the old value.
1. Reset the environment variable to some constant string before freeing the old value.
1. Use setenv() instead of putenv(). Some manual pages recommend this approach.

Reply:
I understand the issue. The free is just before exiting the app, so probably more cosmetic to get a clean valgrind output..